testsuite/tests/typing-poly-bugs/pr5673_bad.compilers.reference

```diff
@@ -3,9 +3,10 @@ Error: This expression has type
          refer1 = < poly : 'a 'b 'c. ('b, 'c) #Classdef.cl2 as 'a >
        but an expression was expected of type
          refer2 = < poly : 'd 'b 'c. ('b, 'c) #Classdef.cl2 as 'd >
-       Type
-         ('b, 'c) Classdef.cl1 =
-           < m : 'b -> 'c -> int; raise_trouble : int -> 'b >
+       Type ('b, 'c, ('b, 'c) Classdef.cl1) Classdef.cl0 = <  >
        is not compatible with type
-         < m : 'b -> 'c -> int; raise_trouble : int -> 'b > 
-       The type variable 'e occurs inside 'e
+         ('b0, 'c0, ('b0, 'c0) Classdef.cl1) Classdef.cl0 
+       Type < m : 'b -> 'c -> int; .. > is not compatible with type
+         ('b, 'c) Classdef.cl1 =
+           < m : 'b -> 'c -> int; raise_trouble : int -> 'b > 
+       The universal variable 'b would escape its scope
```

Comment:
Again, it is very hard to make sense of this error message (`refer1` and `refer2` seems identical modulo `'a -> 'd renaming`, and it is unclear to me why `raise_trouble` makes `'b` escape its scope), but I guess there is little to be done about it. Maybe expect-tests would be slightly less confusing.

Reply:
It helps to refer to the [mantis ticket](https://caml.inria.fr/mantis/view.php?id=5673), which explains that this is a problem with the escape check for universal type variable. In other words, the error is wrong, but at least the error message now points to the source of the problem.